### PR TITLE
ENH: Raw Pointer for Python Wrapping

### PR DESCRIPTION
**Pull request by PranjalSahu**
Minor changes for better code readability and for optimized Python wrapping.
Using Raw Pointer instead of Smart Pointer.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [X] Added test (or behavior not changed)
- [X] Updated API documentation (or API not changed)
- [X] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [X] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [X] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by PranjalSahu** [comment]
@thewtex I didn't do the change to convert K and dA variables to Double vector container as in the loop we are accessing values at various indexes  in non-serial order and using GetElement/SetElement is going to be inefficient in such case (function call stack). Please let me know if there is better alternative.

**Comment by PranjalSahu** [comment]
Should I create separate commits for these changes ?

**Comment by dzenanz** [comment]
You can. Maybe into 3? "Use raw pointer", "Add testing", "Fix wrapping", not necessarily in this order.

File: Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.h [hunk 1]
```diff
@@ -31,6 +31,7 @@
 
 #include <iostream>
 #include "itkMesh.h"
+#include "itkTriangleCell.h"
```
Comment (dzenanz) [inline Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.h:34]: When you have an "**and**" in commit message, it is a good indication it should be split into two commit. There is no limit how small commits can be. I do realize this is nitpicking, so feel free to ignore this time.

**Review by dzenanz — APPROVED** [review]

File: Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.h [hunk 2]
```diff
@@ -129,15 +131,12 @@ class ITK_TEMPLATE_EXPORT TriangleMeshCurvatureCalculator : public Object
 
   /** Discrete Gauss curvature (K) computation */
   void
-  GetGaussCurvature(MeshConstPointer input);
```
Comment (thewtex) [inline Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.h:132]: We want to remove the smart pointer from the interface here.

File: Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.h [hunk 3]
```diff
@@ -129,15 +131,12 @@ class ITK_TEMPLATE_EXPORT TriangleMeshCurvatureCalculator : public Object
...
-
-  void
-  ComputeGaussCurvature(MeshConstPointer input);
+  ComputeGaussCurvature(const InputMeshType * input);
 
 private:
   CurvaturesEnum               m_CurvatureType = CurvaturesEnum::GaussCurvature;
   DoubleVectorContainerPointer m_GaussCurvatureData;
-  MeshConstPointer             m_TriangleMesh;
```
Comment (thewtex) [inline Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.h:140]: This still needs to be a smart pointer (we want to hold a reference to the object).

**Review by thewtex — CHANGES_REQUESTED** [review]
@PranjalSahu thanks! Follow-up noted inline.

**Review by dzenanz — APPROVED** [review]

**Comment by jhlegarreta** [comment]
@PranjalSahu one of these two commits causes a memory leak 59d95db 4aab01e:
https://open.cdash.org/viewDynamicAnalysis.php?buildid=7611627

**Comment by PranjalSahu** [comment]
I will check it. It is probably due to the newly added check for Triangle
Cell. I have to see whether the memory allocated for K and dA are properly
released after the itkExceptionMacro is called.

**Comment by PranjalSahu** [comment]
I think it is due to the TetrahedronCell inserted in the mesh. 
The documentation for [SetCell](https://itk.org/Doxygen/html/classitk_1_1Mesh.html#af48765961f72f75c8d2c6a1d12cd0124) in itk mesh says:

**Access routines to fill the Cells container (m_CellsContainer), and get information from it. If SetCell is used to overwrite a cell currently in the mesh, it is the caller's responsibility to release the memory for the cell currently at the CellIdentifier position prior to calling SetCell.**

So we need to the release the memory of the original cell present in the mesh before setting the new TetrahedronCell.
Testing it and will update here.

**Pull request merged**